src/virt.rs

```diff
@@ -1001,7 +1002,22 @@ impl HwRegisterContextSetter<Csr> for VirtContext {
                     (value & arch_misa & misa::MISA_CHANGE_FILTER & !misa::DISABLED) | misa::MXL;
             }
             Csr::Mie => self.csr.mie = value & hw.interrupts & mie::MIE_WRITE_FILTER,
-            Csr::Mip => self.csr.mip = value & hw.interrupts & mie::MIP_WRITE_FILTER,
+            Csr::Mip => {
+                let value = value & hw.interrupts & mie::MIP_WRITE_FILTER;
+
+                if (self.csr.mip ^ value) & mie::SEIE_FILTER != 0 {
+                    if value & mie::SEIE_FILTER == 0 {
+                        unsafe {
+                            Arch::clear_csr_bits(Csr::Mip, mie::SEIE_FILTER);
+                        }
+                    } else {
+                        unsafe {
+                            Arch::set_csr_bits(Csr::Mip, mie::SEIE_FILTER);
+                        }
+                    }
+                }
+                self.csr.mip = value;
+            }
```

Comment:
We also need some good commenting here, why are we doing that? Explain a bit that there are two SEIE bits in different places and that we need to keep the hardware in sync and how you do it.